File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/watchproviders/model/WatchProviderRegions.java
package dev.reuss.tmdb.domain.watchproviders.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Available TMDB watch provider regions.
 *
 * @param results available regions
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record WatchProviderRegions(
        List<WatchProviderRegion> results
) implements TmdbModel {

    public WatchProviderRegions {
        results = TmdbCollections.list(results);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/watchproviders/model/WatchProviderRegions.kt
package dev.reuss.tmdb.domain.watchproviders.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Available TMDB watch provider regions.
 *
 * @property results available regions
 */
@JvmRecord
data class WatchProviderRegions(
    @all:JsonProperty("results")
    val results: List<WatchProviderRegion> = emptyList(),
) : TmdbModel
